hazelcast-code-generator/src/main/resources/codec-template-java.ftl

```diff
@@ -141,35 +141,45 @@ public final class ${model.className} {
 
 </#list>
 
-
   public static abstract class AbstractEventHandler{
 
         public void handle(ClientMessage clientMessage) {
             int messageType = clientMessage.getMessageType();
         <#list model.events as event>
             if (messageType == com.hazelcast.client.impl.protocol.constants.EventMessageConst.EVENT_${event.name?upper_case}) {
-            boolean messageFinished = false;
+            <#assign paramCallList="">
+            <#assign previousVersion = event.sinceVersion?replace('.','') >
             <#list event.eventParams as p>
-                <@defineVariable var_name=p.name type=p.type />
                 <#if p.versionChanged >
-                    if (!messageFinished) {
-                        messageFinished = clientMessage.isComplete();
+                    if (clientMessage.isComplete()) {
+                        handle${event.name?cap_first}EventV${previousVersion}(${paramCallList});
+                        return;
                     }
                 </#if>
-                    if (!messageFinished) {
-                        <@readVariable var_name=p.name type=p.type isNullable=p.nullable isEvent=true />
-                    }
+                <#if p_index gt 0 ><#assign paramCallList=paramCallList + ", "></#if>
+                <#assign paramCallList += p.name>
```

Comment:
We need following right after this line 
<#assign previousVersion = p.sinceVersion?replace('.','') >

It is available in C# part, looks like missed when porting to here.